### PR TITLE
Add group_other option to fct_collapse, to allow for optional recoding of all non-targeted levels into an "Other" category

### LINKED ISSUE #100: Integrate fct_other functionality into fct_collapse
Thanks for this great package! I'm wondering though whether there would be any possibility of integrating features of `fct_other` into the `fct_collapse` function?

I currently need to recode a factor by collapsing a targeted set of levels (say `c("A", "B", "C", "D", "E")`) into new levels (say `c("AB", "CDE")`) and I want to recode all non-targeted levels (which usually is quite a long set of levels to manually specify, e.g. all other letters of the alphabet) into an `"other"` category. In other words: I'ld like `fct_collapse` to have an additional argument (similar to `fct_other`'s `other_level` argument) that, when specified, automatically identifies all non-targeted levels (i.e. those not specified in any of the vectors, which in `fct_other` would be specified in the `drop` argument) and collapses these into a category with level name specified in this argument.

This feature would be quite helpful, because the only solution I can think of now is quite cumbersome. That is, I have to manually specify all new (collapsed) factor level names (entered in the `fct_collapse` call) in the `keep` argument of `fct_other`, while I assume these could easily be internally identified automatically in the `fct_collapse` function.

### PR DESCRIPTION
**Pull request by AmeliaMN**
Fixes #100

### CODE REVIEW

**Comment by hadley** [comment]
Can you please give this PR an informative title, and then include `Fixes #100` in the description?

**Comment by AmeliaMN** [comment]
Done!

File: R/collapse.R [hunk 1]
```diff
@@ -15,10 +16,16 @@
 #'   dem = c("Not str democrat", "Strong democrat")
 #' )
 #' fct_count(partyid2)
-fct_collapse <- function(.f, ...) {
+fct_collapse <- function(.f, ..., group_other = FALSE) {
   new <- rlang::dots_list(...)
-
   levs <- as.list(unlist(new, use.names = FALSE))
+  if(group_other){
```
Comment (hadley) [inline R/collapse.R]: Missing space before `(`

File: tests/testthat/test-collapse.R [hunk 2]
```diff
@@ -21,3 +21,17 @@ test_that("can collapse missing levels", {
 
   expect_equal(f2, factor(c("x", "y")))
 })
+
+test_that("can collapse un-named levels to Other", {
+  f1 <- factor(letters[1:3])
+  f2 <- fct_collapse(f1, xy = c("a", "b"), group_other=TRUE)
```
Comment (hadley) [inline tests/testthat/test-collapse.R]: Can you please put spaces around `=`?

**Comment by hadley** [comment]
Looks good!

Can you please add a bullet to NEWS? It should briefly [describe the change](http://style.tidyverse.org/news.html) and end with `(@yourname, #issuenumber)`.

**Comment by hadley** [comment]
Thank you!!

**Pull request merged**